### PR TITLE
Try to fix comprehensive tests.

### PR DESCRIPTION
**Pull request by Carreau**
I'm not sure why ti fails, tox says

> unknown environment 'py310-linux-pyside'

While the same env works on test-pull-requests.

So I'mmostly trying to update the comprehensive workflow file to looks
more like the test_pull_requests one, so it's easier to spot differences
that could be the reason for the error.

### CODE REVIEW

**Comment by Czaki** [comment]
It happens because #4646 names are changed from `pyside` to `pyside2` and from `pyqt` to `pyqt5`

**Comment by Carreau** [comment]
Ah ! DUH.

File: .github/workflows/test_comprehensive.yml [hunk 1]
```diff
@@ -40,15 +40,15 @@ jobs:
           # test big sur on 3.9.0
           - python: 3.9
             platform: macos-11
-            toxenv: py39-macos-pyqt
+            backend: pyqt
```
Comment (Carreau) [inline .github/workflows/test_comprehensive.yml]:
```suggestion
            toxenv: py39-macos-pyqt2
```

File: .github/workflows/test_comprehensive.yml [hunk 2]
```diff
@@ -40,15 +40,15 @@ jobs:
...
           # test with minimum specified requirements
           - python: 3.8
             platform: ubuntu-18.04
             MIN_REQ: 1
           # test with --async_only
           - python: 3.8
             platform: ubuntu-18.04
-            toxenv: async-py38-linux-pyqt
+            backend: pyqt
```
Comment (Carreau) [inline .github/workflows/test_comprehensive.yml]:
```suggestion
            async-py38-linux-pyqt2
```

File: .github/workflows/test_comprehensive.yml [hunk 3]
```diff
@@ -40,15 +40,15 @@ jobs:
           # test big sur on 3.9.0
           - python: 3.9
             platform: macos-11
-            toxenv: py39-macos-pyqt
+            toxenv: py39-macos-pyqt2
```
Comment (tlambert03) [inline .github/workflows/test_comprehensive.yml:43]: I assume this is a typo.  (would need to be `pyqt5`, not `2`)
each item in that list needs to match one of the factors in the envlist at the top of tox.ini 

anyway, this is fixed in #4660  ... apologies for not seeing this PR

**Pull request closed without merging**